cls/SourceControl/Git/Extension.cls

```diff
@@ -236,15 +254,17 @@ Method OnAfterSave(InternalName As %String, Object As %RegisteredObject = {$$$NU
 {
     set sc = $$$OK
     try {
-        set InternalName = ##class(Utils).NormalizeInternalName(.InternalName)
+        set InternalName = ##class(SourceControl.Git.Utils).NormalizeInternalName(.InternalName)
         set context = ##class(SourceControl.Git.PackageManagerContext).ForInternalName(InternalName)
-        if ##class(Utils).IsNamespaceInGit() && ..IsInSourceControl(InternalName) {
-            set filename = ##class(Utils).FullExternalName(InternalName)
-            $$$ThrowOnError(##class(Utils).RemoveRoutineTSH(InternalName))
-            $$$ThrowOnError(##class(Utils).ExportItem(InternalName))
+        if ##class(SourceControl.Git.Utils).IsNamespaceInGit() && ..IsInSourceControl(InternalName) {
+            set filename = ##class(SourceControl.Git.Utils).FullExternalName(InternalName)
+            $$$ThrowOnError(##class(SourceControl.Git.Utils).RemoveRoutineTSH(InternalName))
+            $$$ThrowOnError(##class(SourceControl.Git.Utils).ExportItem(InternalName))
             if '##class(SourceControl.Git.Change).IsUncommitted(filename) {
                 $$$ThrowOnError(##class(SourceControl.Git.Change).SetUncommitted(filename, "edit", InternalName, $username, "", 1, "", "", 0))
             }
+        } elseif ##class(SourceControl.Git.Utils).BasicMode() {
+            do ..AddToSourceControl(InternalName)
```

Comment:
I think this belongs in UserAction/AfterUserAction for creation of new things (those are types of system events) rather than OnAfterSave - if a user elects to remove (not track) a certain file we should reflect that.

Reply:
Right. You make a good point here! Not sure what I was thinking at the time.